azure_functions_worker/dispatcher.py

```diff
@@ -313,82 +322,92 @@ async def _handle__worker_status_request(self, request):
             request_id=request.request_id,
             worker_status_response=protos.WorkerStatusResponse())
 
-    async def _handle__functions_metadata_request(self, request):
-        metadata_request = request.functions_metadata_request
-        directory = metadata_request.function_app_directory
+    def get_function_metadata(self, directory, caller_info):
+        """
+        This method is called to index the functions in the function app
+        directory and save the results in function_metadata_result or
+        function_metadata_exception in case of an exception.
+        """
         script_file_name = get_app_setting(
             setting=PYTHON_SCRIPT_FILE_NAME,
             default_value=f'{PYTHON_SCRIPT_FILE_NAME_DEFAULT}')
-        function_path = os.path.join(directory, script_file_name)
 
         logger.info(
-            'Received WorkerMetadataRequest, request ID %s, function_path: %s',
-            self.request_id, function_path)
+            'Received WorkerMetadataRequest from %s, request ID %s, '
```

Comment:
nit; this is not always called from workermetadatarequest

Reply:
Its not. But it still to get the metadata request. The caller will be logged in caller_info.

Reply:
Actually it does help us finding out from where the request is coming from. This would be good for telemetry. Also it wont add any additional latency than what we already have. This will be logged only once, either from the init/reload request or the metadata request, 
I can change the log statement to read 'Received load metadata request from  .....'

Reply:
Fixed. added original log to metadata request